src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -23,28 +23,37 @@
 # pylint: disable=inconsistent-return-statements
 def _query_account_key(cli_ctx, account_name):
     """Query the storage account key. This is used when the customer doesn't offer account key but name."""
+    rg, scf = _query_account_rg(cli_ctx, account_name)
+    t_storage_account_keys, t_storage_account_list_keys_results = get_sdk(
+        cli_ctx,
+        ResourceType.MGMT_STORAGE,
+        'models.storage_account_keys#StorageAccountKeys',
+        'models.storage_account_list_keys_result#StorageAccountListKeysResult')
+
+    if t_storage_account_keys:
+        return scf.storage_accounts.list_keys(rg, account_name).key1
+    elif t_storage_account_list_keys_results:
+        return scf.storage_accounts.list_keys(rg, account_name).keys[0].value  # pylint: disable=no-member
```

Comment:
Then if either of the types exists, this method returns `None`? Expected?

Reply:
Currently for all the api-versions, it's either or.
To prevent possible bugs with new api-versions in the future, I'll make this an `else`, which will throw an error when it tries to get the first key.